pipeline/src/test_util.rs

```diff
@@ -532,3 +532,55 @@ pub fn run_reparse_test_with_blacklist(file: &str, blacklist: &[&str]) {
         .compute_optimized_pil()
         .unwrap();
 }
+
+// TODO: Add #[cfg(feature = "stwo")] to conditionally compile this function. now when I add it, this code is disabled. how to eable by default?
+#[cfg(feature = "stwo")]
+use powdr_number::Mersenne31Field;
+#[cfg(feature = "stwo")]
+pub fn test_stwo(file_name: &str, inputs: Vec<Mersenne31Field>) {
+    let backend = powdr_backend::BackendType::Stwo;
+    let mut pipeline = Pipeline::default()
+        .with_tmp_output()
+        .from_file(resolve_test_file(file_name))
+        .with_prover_inputs(inputs.clone())
+        .with_backend(backend, None)
+        .set_witness(vec![
+            (
+                "add::a".to_string(),
+                vec![
+                    Mersenne31Field::from(1),
+                    Mersenne31Field::from(2),
+                    Mersenne31Field::from(1),
+                    Mersenne31Field::from(2),
+                ],
+            ),
+            (
+                "add::b".to_string(),
+                vec![
+                    Mersenne31Field::from(1),
+                    Mersenne31Field::from(2),
+                    Mersenne31Field::from(1),
+                    Mersenne31Field::from(2),
+                ],
+            ),
+            (
+                "add::c".to_string(),
+                vec![
+                    Mersenne31Field::from(2),
+                    Mersenne31Field::from(4),
+                    Mersenne31Field::from(2),
+                    Mersenne31Field::from(4),
+                ],
+            ),
+        ]);
+    // Generate a proof
+    let proof = pipeline.compute_proof().cloned().unwrap();
```

Comment:
```suggestion
    let proof = Pipeline::default()
        .with_tmp_output()
        .from_file(resolve_test_file(file_name))
        .with_prover_inputs(inputs.clone())
        .with_backend(backend, None)
        .compute_proof().cloned().unwrap();
```

this might work, the witness gets computed if needed

Reply:
it works, but it always set witness to be all 0, would this be enough for test case?

Reply:
Right, so this is due to the strategy witness generation uses. It tries to find a unique witness on each row by solving the constraints. However, in the case where it cannot find a unique solution, it makes a last attempt setting everything to zero. If the constraints are satisfied, it outputs this witness.

This program does not have a unique witness, so it applies this strategy and sets everything to zero.

I think that's fine for this test, we'll have less trivial examples once we support fixed columns. Also we already have an `add` test in `pipeline/tests/pil` so we should add stwo there and we can remove this new test.

Reply:
I see. make sense. witness generation is automatic now.